src/components/App.jsx

```diff
@@ -0,0 +1,135 @@
+import { useEffect, useState } from 'react';
+
+import Header from './Header.jsx';
+import Main from './Main.jsx';
+import Sidebar from './Sidebar.jsx';
+import LoginModal from './LoginModal.jsx';
+import SignupModal from './signupModal.jsx';
+import PreviewModal from './PreviewModal.jsx';
+import GoogleBooksSearch from './Views/ActiveBooklistView.jsx';
+import { getBook, getBestSellers } from '../utils/api.jsx';
+
+import {
+  screenWidthContext,
+  searchResultsContext,
+} from '../contexts/contexts.js';
+
+export default function App() {
+  const [screenWidth, setScreenWidth] = useState(window.innerWidth);
+  const [searchResults, setSearchResults] = useState([]);
+  const [activeModal, setActiveModal] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedBook, setSelectedBook] = useState({});
+  const [shelves, setShelves] = useState([]);
+
+  const handleCloseModal = () => {
+    setActiveModal('');
+  };
+
+  const handleLogin = () => {
+    console.log('logged in');
+  };
+
+  const handleSignup = () => {
+    console.log('signed up');
+  };
+
+  const handleSignupClick = () => {
+    setActiveModal('signup');
+  };
+
+  const handleLoginClick = () => {
+    setActiveModal('login');
+  };
+
+  const handleCardClick = (book) => {
+    console.log(book);
+    setActiveModal('preview');
+    setSelectedBook(book);
+  };
+
+  const toPascalCase = (str) => {
+    return str
+      .toLowerCase()
+      .split(/[\s-_]+/)
+      .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+      .join(' ');
+  };
+
+  useEffect(() => {
+    window.addEventListener('resize', (e) => {
+      setScreenWidth(e.target.innerWidth);
+    });

```

Comment:
Please, don’t add listeners to `document` or `window` if you don’t remove them because the listeners always watch any clicks on the page and make checks. That’s a waste of resources.


And there will be **memory leak** if `App` rerenders


Please, add the function correctly, and remove it in `clean up`

Reply:
[feat: update responsive design functionality to remove listener] - I opted to get rid of the listener all together and replaced it with a ResizeObserver. This also removed the ScreenWidthContext. This refactor is a lot cleaner and more consise